### PR TITLE
Standalone web request tracing

### PR DESCRIPTION
**Pull request by stefaneberl**
Well not completely standalone, a Session is still required. :)

Also refactored some existing tests, since it was not quite cool the way
it was originally written.

### CODE REVIEW

File: src/main/java/com/dynatrace/openkit/core/SessionImpl.java [hunk 1]
```diff
@@ -16,15 +16,13 @@
 
 package com.dynatrace.openkit.core;
 
-import com.dynatrace.openkit.api.Action;
-import com.dynatrace.openkit.api.Logger;
-import com.dynatrace.openkit.api.RootAction;
-import com.dynatrace.openkit.api.Session;
+import com.dynatrace.openkit.api.*;
```
Comment (y0chanan) [inline src/main/java/com/dynatrace/openkit/core/SessionImpl.java]: can this pollute the imports, referencing stuff that was not intended? I would prefer 4 explicitly defined imports

File: src/test/java/com/dynatrace/openkit/core/SessionImplTest.java [hunk 2]
```diff
@@ -493,44 +453,187 @@ public void enterActionGivesNullRootActionIfSessionIsAlreadyEnded() {
...
+        target.end();
+
+        // when
+        WebRequestTracer obtained = target.traceWebRequest("http://www.google.com");
+
+        // then
+        assertThat(obtained, is(notNullValue()));
+        assertThat(obtained, is(instanceOf(NullWebRequestTracer.class)));
+    }
+
```
Comment (y0chanan) [inline src/test/java/com/dynatrace/openkit/core/SessionImplTest.java]: unnecessary lines from here on

File: src/main/java/com/dynatrace/openkit/core/SessionImpl.java (hunk 1 above)
Comment (stefaneberl) [reply src/main/java/com/dynatrace/openkit/core/SessionImpl.java]: Basically, it's 5 imports, which is the default threshold for IntelliJ to use wildcard imports.

I've changed this in my IDE.

File: src/main/java/com/dynatrace/openkit/api/Session.java [hunk 3]
```diff
@@ -62,6 +63,27 @@
...
+    /**
+     * Traces a web request - which is provided as a URLConnection - and allows adding timing information to this request.
+     * If the web request is continued on a server-side Agent (e.g. Java, .NET, ...) this Session will be correlated to
+     * the resulting server-side PurePath.
+     *
+     * @param connection the URLConnection of the HTTP request to be tagged and timed
+     * @return a WebRequestTracer which allows adding timing information
+     */
+    WebRequestTracer traceWebRequest(URLConnection connection);
+    /**
```
Comment (thomasgrassauer) [inline src/main/java/com/dynatrace/openkit/api/Session.java]: add linebreak before doc

**Pull request merged**